src/python/pants/core/util_rules/system_binaries.py

```diff
@@ -454,7 +491,11 @@ async def get_bash() -> BashBinary:
 
 
 @rule
-async def find_binary(request: BinaryPathRequest, env_target: EnvironmentTarget) -> BinaryPaths:
+async def find_binary(
+    request: BinaryPathRequest,
+    env_target: EnvironmentTarget,
+    # system_binaries: SystemBinariesSubsystem,
```

Comment:
Remove this commented-out line (and then restore the formatting so this passes lint)

Reply:
But note that this rule does need some work: Currently it will always append to the real PATH (via the call to `create_path_env_var`), but now that we allow that to be configurable, we should only use the real PATH if requested.

Reply:
Looks good now, but I've noticed that `create_path_env_var` is now never correct! All the places we use it support the `<PATH>` special var, and `create_path_env_var` will add the PATH when we don't want or need it to. Seems like a bug to. me.

Fixing that should probably be a followup, so we can get this one merged now. But could you file a bug for it? And, ideally, fix it in a followup? :) Thanks!

Reply:
I don't know if I can file a bug for that because I'm not quite sure what you mean. From how I understand the code `create_path_env_var` will only add the system PATH if it's passed inside the `env: dict[str, str] | None` parameter, and in the places it's used it's never passed more than one argument, `new_entries`, a string iterator it joins with a delimiter.

Reply:
Ah yes, it's not really a bug at the moment, you're right, since nowhere provides an env. It's just unnecessary, because all it does is split and rejoin the path. But if someone were to pass an env, then it would actually be a bug.

Reply:
Fixed here: https://github.com/pantsbuild/pants/pull/20004